source/MRVoxels/MRMinimalSurface.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <MRMesh/MRMeshFwd.h>
+#include <MRMesh/MRExpected.h>
+#include <MRVoxels/MRVoxelsFwd.h>
+
+
+namespace MR
+{
+
+
+/// Supported types of TPMS (Triply Periodic Minimal Surfaces)
+enum class TPMSType : int
+{
+    SchwartzP,
+    DoubleSchwartzP,
+    Gyroid,
+    DoubleGyroid,
+};
+MRVOXELS_API std::vector<std::string> getTPMSTypeNames();
+
+using TPMSFunction = float(*)( const Vector3f& );
+
+/// Get TPMS by its type
+MRVOXELS_API TPMSFunction getTPMSFunction( TPMSType type );
+
+
+/// Construct TPMS using implicit function (https://www.researchgate.net/publication/350658078_Computational_method_and_program_for_generating_a_porous_scaffold_based_on_implicit_surfaces)
+/// @param type Type of the surface
+/// @param size Size of the cube with the surface
+/// @param frequency Frequency of oscillations (determines size of the "cells" in the "grid")
+/// @param resolution Ratio `n / T`, between the number of voxels and period of oscillations
+/// @return Distance-volume starting at (0, 0, 0) and having specified @p size
+MRVOXELS_API FunctionVolume buildTPMSVolume( TPMSType type, const Vector3f& size, float frequency, float resolution );
+
+/// Constructs TPMS level-set and then convert it to mesh
+MRVOXELS_API Expected<Mesh> buildTPMSSurface( TPMSType type, const Vector3f& size, float frequency, float resolution, float iso );
+
+/// Constructs TPMS-filling for the given @p mesh
+MRVOXELS_API Expected<Mesh> buildTPMSSurface( TPMSType type, const Mesh& mesh, float frequency, float resolution, float iso );
```

Comment:
better give another name to the function since it does not do the same as above function, e.g. `fillWithTpmSponge`